src/components/select/__internal__/utils/highlight-part-of-text.tsx

```diff
@@ -29,7 +36,9 @@ function highlightPartOfText(
 
   const newValue = [
     <span key="preceding">{precedingText}</span>,
-    <MatchingText key="match">{matchingText}</MatchingText>,
+    <span key="match" data-role="matching-text" style={highlightStyling}>
```

Comment:
**comment** (non-blocking): I'm not sure why you've replaced the styled component here with inline styles. I'm confident it will have the same effect but I'd prefer to avoid inline styles unless we have a really good reason, and while I may be missing something, I don't see any reason here to make this change. (I know you've added the `data-role` as I requested, but as a `data-` attribute it ought to be fine for any HTML element, and even if TS does complain for some reason you could just have explicitly added `data-role` to the props interface for `MatchingText`.)